src/app/courses/step-view-courses/courses-step-view.component.ts

```diff
@@ -177,4 +178,36 @@ export class CoursesStepViewComponent implements OnInit, OnDestroy {
     });
   }
 
+  surveyExists(): boolean {
```

Comment:
In addition, `surveyExists` and `examExists` can be refactored into one function that takes a string `'exam' | 'survey'`